isce2_topsapp/localize_slc.py

```diff
@@ -238,3 +229,33 @@ def download_one(resp):
         "reference_properties": reference_props,
         "secondary_properties": secondary_props,
     }
+
+
+def _get_frame_by_id(frame_id: int) -> gpd.GeoSeries:
+    frames = gpd.read_file(Path(__file__).parent / 'data' / 's1_frames_latitude_aligned.geojson.zip')
+    return frames[frames.frame_id == frame_id].reset_index(drop=True).iloc[0]
+
+
+def _get_dates(product: asf.ASFProduct) -> set[datetime.date]:
+    date_strings = [product.properties[field].strip('Z') for field in ['startTime', 'stopTime']]
```

Comment:
If I delete `startTime` from this list, the test fails at the assertion on L192, as expected. But if I delete `stopTime`, then the test still passes. So it might be good to find a test case for crossing midnight where only `stopTime` falls on the given date.